### DISCUSSION TITLE
Working with replicated point patterns

### DISCUSSION THREAD

**Discussion by daniel-manrique** [inlabru-org/inlabru#183] [General]
Dear,

Lately, I have been working on the analysis of point patterns. I am a neurobiologist, and I have dozens of point patterns showing the position of cells in the brain. Is there an effective way to work with replicated point patterns using inlabru? 

Regards

**Comment by finnlindgren** [comment]
Hi, this depends on more of the details of what kind of model you want to have. Are the point patterns replicated at the point level, with a common intensity function across replications, or does each point pattern involve its own (randomly replicated) intensity field? Also, is it flat 2D space, 3D space, or curved 2-manifold space that the points "live" on?

**Reply by daniel-manrique** [reply]
I appreciate your response. I have randomly replicated points with their own intensity fields (also nested in different experimental groups or conditions). They are 2d. But now that you mention it, I will have in the next months 3d assemblies 😃😃. The way I'm working with them now is using the hiperframes manipulations allowed by spatstat for replicated point patterns. However, I would like to go bayesian, and overall, overpass the limitations spatstat have with replicated point patterns.

Thank you

**Reply by finnlindgren** [reply]
I'm not sure what `spatstat` can and can't do with replicated point patterns, but it sounds like you want a model where the intensity field is different for each point pattern, but they might have the same covariance parameters in common?

That case can be formulated as a random field model with independent replicates, that's then observed via a point pattern model. One way of specifying that might look something like this:
```r
# assume `points` are the points stored as an `sf` object, with an additional `replicate` column
# indicating which replicate each point was observed in.
spde <- inla.spde2.pcmatern(mesh, ...)
components <- ~ Intercept(1) + field(geometry, model = spde, replicate = replicate)
lik <- like(
  formula = geometry + replicate ~ Intercept + field,
  data = points,
  domain = list(geometry = mesh, replicate = INLA::inla.mesh.1d(1:N_replicates, indexed = TRUE)))
fit <- bru(components, lik)
```
